File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/board/service/CreateBoard.java
package com.example.mediumspringwebflux.board.service;

import com.example.mediumspringwebflux.board.dto.BoardRequest;
import com.example.mediumspringwebflux.board.document.Board;
import com.example.mediumspringwebflux.board.repository.BoardRepository;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class CreateBoard {

    private final BoardRepository boardRepository;

    public CreateBoard(BoardRepository boardRepository) {
        this.boardRepository = boardRepository;
    }

    public Mono<Void> execute(BoardRequest request) {
        return Mono.just(Board.builder()
                        .title(request.title())
                        .description(request.description())
                        .build())
                .flatMap(boardRepository::save)
                .then();
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/service/CreateBoard.kt
package com.example.mediumspringwebfluxcorutine.board.service

import com.example.mediumspringwebfluxcorutine.board.document.Board
import com.example.mediumspringwebfluxcorutine.board.dto.BoardRequest
import com.example.mediumspringwebfluxcorutine.board.repository.BoardRepository
import kotlinx.coroutines.reactor.awaitSingle
import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
class CreateBoard(
    private val boardRepository: BoardRepository
) {

    suspend fun execute(boardRequest: BoardRequest) {
        boardRepository.save(Board(
                title = boardRequest.title,
                content = boardRequest.content,
                createdDate = LocalDateTime.now(),
                updatedDate = LocalDateTime.now()))
            .awaitSingle()
    }

}